src/main/java/org/broadinstitute/hellbender/tools/funcotator/dataSources/gencode/GencodeFuncotationFactory.java

```diff
@@ -1077,47 +1086,53 @@ private GencodeFuncotation createUtrFuncotation(final VariantContext variant,
+                //       even if the variant occurs in the last base of the UTR.
+                final int numExtraTrailingBases = variant.getReference().length() < defaultNumTrailingBasesForUtrAnnotationSequenceConstruction ? defaultNumTrailingBasesForUtrAnnotationSequenceConstruction : variant.getReference().length() + 1;
+                final String fivePrimeUtrCodingSequence =
+                        getFivePrimeUtrSequenceFromTranscriptFasta( transcript.getTranscriptId(), transcriptIdMap, transcriptFastaReferenceDataSource, numExtraTrailingBases);
+
+                final int codingStartPos = FuncotatorUtils.getStartPositionInTranscript(variant, activeRegions, strand);
+
+                // But we can really just use the referenceBases to do this:
+                final String rawAltUtrSubSequence = (referenceBases.substring(referenceWindow-numLeadingBasesForUtrAnnotationSequenceConstruction, referenceWindow) +
+                        strandCorrectedAltAllele +
+                        referenceBases.substring(referenceWindow + variant.getReference().length(), referenceWindow + numExtraTrailingBases));
+
+                // Check for de novo starts in the raw sequence:
+                boolean startFound = false;
+                int codingRegionOffset = -numLeadingBasesForUtrAnnotationSequenceConstruction;
+                for ( int i = 0; (i+3 < rawAltUtrSubSequence.length()) ; ++i ) {
+                    startFound = FuncotatorUtils.getEukaryoticAminoAcidByCodon( rawAltUtrSubSequence.substring(i, i+3) ) == AminoAcid.METHIONINE;
+                    if (startFound) {
+                        codingRegionOffset += i;
+                        break;
+                    }
                 }
-                else {
-                    gencodeFuncotationBuilder.setVariantClassification(GencodeFuncotation.VariantClassification.DE_NOVO_START_OUT_FRAME);
+                // If we found a start codon, we should set the variant classification as such:
+                if ( startFound ) {
+                    if ( FuncotatorUtils.isInFrameWithEndOfRegion(codingStartPos + codingRegionOffset, fivePrimeUtrCodingSequence.length()) ) {
+                        gencodeFuncotationBuilder.setVariantClassification(GencodeFuncotation.VariantClassification.DE_NOVO_START_IN_FRAME);
+                    }
+                    else {
+                        gencodeFuncotationBuilder.setVariantClassification(GencodeFuncotation.VariantClassification.DE_NOVO_START_OUT_FRAME);
+                    }
                 }
             }
-            // Just a normal 5' variant:
             else {
-                gencodeFuncotationBuilder.setVariantClassification(GencodeFuncotation.VariantClassification.FIVE_PRIME_UTR);
+                logger.warn("Attempted to process transcript information for transcript WITHOUT sequence data.  Ignoring sequence information for Gencode Transcript ID: " + transcript.getTranscriptId());
```

Comment:
When would this happen?  Not sure if any action is needed.

Reply:
This would happen when the Gencode data source does not have a sequence for a given transcript.  The information in the resulting funcotation will be correct, but it will be a subset of what a user may expect.  This does happen in practice, so I wanted to make sure it was logged as a warning.